test/common/router/BUILD

```diff
@@ -280,8 +280,13 @@ envoy_cc_test(
 
 envoy_cc_test(
     name = "router_test",
-    srcs = ["router_test.cc"],
+    srcs = [
+        # Split to avoid compiler OOM, especially on ASAN.
+        "router_test.cc",
+        "router_2_test.cc",
```

Comment:
don't we need 2 test targets to run them individually?

Reply:
The problem occurs during compilation, because the file grew too large (~6K SLOC), not running the test, which is fast. I could split it into two test targets but I'm unsure how it will affect corpus generation.

Reply:
It's `config_impl_test` that generates a corpus, not router_test.

Reply:
in that case WDYT about making 2 test targets, Yan?